allure-support/src/main/kotlin/com/kaspersky/components/alluresupport/interceptors/testrun/VideoRecordingTestInterceptor.kt

```diff
@@ -1,19 +1,40 @@
 package com.kaspersky.components.alluresupport.interceptors.testrun
 
 import com.kaspersky.components.alluresupport.files.attachVideoToAllureReport
+import com.kaspersky.components.alluresupport.files.dirs.AllureDirsProvider
 import com.kaspersky.kaspresso.device.video.Videos
 import com.kaspersky.kaspresso.interceptors.watcher.testcase.TestRunWatcherInterceptor
 import com.kaspersky.kaspresso.testcases.models.info.TestInfo
+import io.qameta.allure.kotlin.Allure
+import io.qameta.allure.kotlin.model.Status
 
+/**
+ * Due to screen recorder bug we have to perform a workaround which requires VideoRecordingTestInterceptor
+ * and MoveReportsInterceptor to be the last two interceptors in allure reports. So if you use VideoRecordingTestInterceptor
+ * be sure to use MoveReportsInterceptor too. Otherwise your report will be under /data/data/your.package.name/files/allure-results
+ * and videos under /sdcard
+ */
 class VideoRecordingTestInterceptor(
-    private val videos: Videos
+    private val videos: Videos,
+    private val allureDirsProvider: AllureDirsProvider,
+    private val stateHolder: AttachedAllureVideosHolder
 ) : TestRunWatcherInterceptor {
 
     override fun onTestStarted(testInfo: TestInfo) {
         videos.record("Video_${testInfo.testName}")
     }
 
     override fun onTestFinished(testInfo: TestInfo, success: Boolean) {
-        videos.saveAndApply { attachVideoToAllureReport() }
+        videos.saveAndApply {
+            val stubFile = allureDirsProvider.provideReportVideoAttachmentStub(this)
+            stubFile.attachVideoToAllureReport()
+            val uuid = Allure.lifecycle.getCurrentTestCase() ?: ""
+            Allure.lifecycle.updateTestCase {
```

Comment:
Seems like these calls to `Allure.lifecycle` are being performed only by `AllureJunit4` and most likely if `AllureRunner` is set by the user. 

What if user have set the `AllureRunner`? I mean we access the other lib's interior lifecycle management manually from here. And we force the test case to stop and to write resources before the allure does so itself. How will this affect the allure's lifecycle?

What if the user have not set the `AllureRunner`? Seems like `startTestCase` is never called but `startTestCase` is.

As far as I understand the intention here is just to link the video file with the test, so can we generate the uuid by our own and manage it separately from allure like we do it in `AllureMapperStepInterceptor`?

Reply:
1. Yes, idea is to make allure write down report so we could parse it and check which UUID it generated for our video file. 
2. Yes, it affects allure's lifecycle - it stops it. I don't think it is a problem either. We interact with allure lifecycle in the end of the test. Each test is run is a separate process, allure doesn't outlive it
3. It would be strange if user wants to launch allure test without configuring allure runner. I don't think It's kaspresso's problem. We could try to detect whether allure runner is used and crash with a verbal message to make user set up runner

Reply:
My bad, it seemed to me `AllureRunner` is optional.

Reply:
Tests are not necessary executed each in its own process, there may be no orchestration. Let me check my concern

Reply:
`updateTestCase`, `stopTestCase`, `writeTestCase` will be called the second time by allure. And the `updateTestCase` contains little bit different logic for status. Will return on this tomorrow, maybe we could avoid this.

Reply:
We could but then we need to provide our own instrumentation runner with the run listener. So yeah seems like this hack is the only way. Also there will be errors printed in logcat by allure after each test saying that the testcase can't be found to be stopped.